crates/control/src/motion/fall_protector.rs

```diff
@@ -1,51 +1,46 @@
-use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use std::{ops::Range, time::Duration};
 
-use approx::relative_eq;
 use color_eyre::Result;
 use context_attribute::context;
-use filtering::low_pass_filter::LowPassFilter;
 use framework::MainOutput;
-use hardware::PathsInterface;
-use motionfile::{MotionFile, MotionInterpolator};
-use nalgebra::Vector2;
 use serde::{Deserialize, Serialize};
 use types::{
-    condition_input::ConditionInput,
     cycle_time::CycleTime,
-    fall_state::FallState,
-    joints::{body::BodyJoints, head::HeadJoints, Joints},
-    motion_command::{FallDirection, MotionCommand},
-    motion_selection::{MotionSafeExits, MotionSelection, MotionType},
+    fall_state::{FallDirection, FallState, Side},
+    joints::{arm::ArmJoints, body::BodyJoints, head::HeadJoints, leg::LegJoints, Joints},
+    motion_selection::{MotionSafeExits, MotionType},
     motor_commands::MotorCommands,
-    parameters::{FallProtectionParameters, FallStateEstimationParameters},
     sensor_data::SensorData,
 };
 
-#[derive(Deserialize, Serialize)]
-pub struct FallProtector {
-    start_time: SystemTime,
-    interpolator: MotionInterpolator<Joints<f32>>,
-    roll_pitch_filter: LowPassFilter<Vector2<f32>>,
-    last_fall_state: FallState,
-    fallen_time: Option<SystemTime>,
+#[derive(Clone, Copy, Debug)]
+enum Phase {
```

Comment:
Isn't that clear as this type is private in the `fall_protector`? This seems to be a violation of *do-not-repeat-yourself*

Reply:
Yes, but it is not the phase of the `FallProtector` but the `Fall`. I just remember being confused because I was unsure what this `Phase` actually refers to.